geth/api/backend.go

```diff
@@ -208,14 +208,46 @@ func (b *StatusBackend) SendTransaction(ctx context.Context, args common.SendTxA
 	return rst.Hash, nil
 }
 
+func (b *StatusBackend) getVerifiedAccount(password string) (*account.SelectedExtKey, error) {
+	selectedAccount, err := b.accountManager.SelectedAccount()
+	if err != nil {
+		b.log.Warn("failed to get a selected account", "err", err)
+		return nil, err
+	}
+	// make sure that only account which created the tx can complete it
+	config, err := b.NodeManager().NodeConfig()
+	if err != nil {
+		return selectedAccount, err
+	}
+	_, err = b.accountManager.VerifyAccountPassword(config.KeyStoreDir, selectedAccount.Address.String(), password)
+	if err != nil {
+		b.log.Warn("failed to verify account", "account", selectedAccount.Address.String(), "error", err)
+		return selectedAccount, err
+	}
+	return selectedAccount, nil
+}
+
 // CompleteTransaction instructs backend to complete sending of a given transaction
 func (b *StatusBackend) CompleteTransaction(id common.QueuedTxID, password string) (gethcommon.Hash, error) {
-	return b.txQueueManager.CompleteTransaction(id, password)
+	selectedAccount, verificationErr := b.getVerifiedAccount(password)
```

Comment:
why you can't exit here if the account is nil or there was a verification err?

Reply:
Sure, I know this can result a bit ugly :(. 

Basically `txQueuemanager` [removes transaction from queue if no error, or error is not transient](https://github.com/status-im/status-go/pull/773/files/ffdbab782fc3a62a6b74766f966b11025993a232#diff-a16061196c3bf37b6badd79a450b352aL146). 

To keep the changes compatible with current behaviour txQueueManager needs to have access to the error generated by the account verification.

Reply:
Sure, I know this can result a bit ugly :(

Basically `txQueuemanager` [removes transaction from queue if no error, or error is not transient](https://github.com/status-im/status-go/pull/773/files/ffdbab782fc3a62a6b74766f966b11025993a232#diff-a16061196c3bf37b6badd79a450b352aL146). 

To keep the changes compatible with current behaviour txQueueManager needs to have access to the error generated by the account verification.

Reply:
Could you add a little comment about it for a later maintainer?

Reply:
right, so 2/3 transient errors can be avoided if we will exit here with an error or if the account is nil (1 and 3 in the list below). but there is still a notification... please check if we can easily move NotifyOnReturn up the stack? If not then fine

```
keystore.ErrDecrypt.Error():          true, // wrong password
ErrInvalidCompleteTxSender.Error():   true, // completing tx create from another account
account.ErrNoAccountSelected.Error(): true, // account not selected
```

Reply:
cheers, i'll have a look

Reply:
I've exposed `TxQueueManager.NotifyErrored` [on this commit](https://github.com/status-im/status-go/pull/773/commits/912f8bda59053f2361fd22c1c1071bb09ab828e2) and linked it from StatusBackend.

Not sure it makes sense use it that way, could you please have a look?